### DISCUSSION TITLE
How does nextjs generate uncompressed html code?

### DISCUSSION THREAD

**Discussion by luojinghui** [vercel/next.js#42045] [Help]
Hello，everyone，by default, nextjs is used to build resources and access pages, the html document content is compressed into one line of code, which displays no problem under the browser, but this is optimized for SEO and requires the use of uncompressed html code content, I try to configure :
```js
swcMinify: false,
compress: false,
reactStrictMode: true,
```
None of them solved my problem, so I want to know if there are other configurations that can avoid html code compression?

## Screenshot of the problem
![image](https://user-images.githubusercontent.com/12367406/198543664-f3d9e965-9896-4537-95ef-e1c6a3b236c3.png)

## Expected performance
![image](https://user-images.githubusercontent.com/12367406/198544142-410fa37b-909a-4d37-8491-ac6d6099748d.png)

Thanks.

**Comment by icyJoseph** [comment]
I think many have tried this and it is not possible.

This one:

```js
const nextConfig = {
  reactStrictMode: true,
  webpack(config, { nextRuntime }) {
    if (!nextRuntime) {
      config.optimization.minimizer = [];
    }
    return config;
  },
};

module.exports = nextConfig;
```

Removes a bunch of minifications, but the HTML string is still the same. I might be talking nonsense here, but I do think the problem is that the rendered string comes form calling render to string, which doesn't add any formatting. So it is not like they are "destructively"  trying to compress the HTML, so much so as, that, the result from walking the JSX and rendering is a very long string of HTML.

**Reply by luojinghui** [reply]
It's too bad, it seems that the html is not processed by Webpack.

I tried to export the static html through `next export`, and then beautify the html through the node script, but it only works for all static pages that are not i18n, otherwise there will be builds mistake.

Or run through a `custom server`, but this requires a separate development of services to implement, which is more troublesome to implement.

**Reply by icyJoseph** [reply]
I think you'll see the problem better if you go to a codesandbox, https://codesandbox.io/s/new?file=/src/index.js, for React and change the `index.js` entry to:

```jsx
import { StrictMode } from "react";
import {renderToString} from "react-dom/server"
import { createRoot } from "react-dom/client";

import App from "./App";

const rootElement = document.getElementById("root");
const root = createRoot(rootElement);

console.log(renderToString(<App/>)) // take a look at this at the console

root.render(
  <StrictMode>
    <App />
  </StrictMode>
);
```

This is an extreme simplification, but you see, the DOM rendered to String is not prettified, rather, it is not formatted, uglyfied or minified at all, it is just a plain string which represents the DOM at the first frame.
